apps/system/lockscreen/js/lockscreen.js

```diff
@@ -851,15 +852,15 @@
    */
   LockScreen.prototype.checkPassCode =
   function lockscreen_checkPassCode(passcode) {
-    var request = {
-      passcode: passcode,
-      onsuccess: this.onPasscodeValidationSuccess.bind(this),
-      onerror: this.onPasscodeValidationFailed.bind(this),
-    };
-    window.dispatchEvent(new CustomEvent(
-      'lockscreen-request-passcode-validate',
-      { detail: request }
-    ));
+    PasscodeHelper.check(passcode).then((result) => {
+      if (result) {
+        this.onPasscodeValidationSuccess();
+      } else {
+        this.onPasscodeValidationFailed();
+      }
+    }) .catch(() => {
```

Comment:
If you catch something, I would expect to see the code handle it:

    .catch((error) => {    this.onPassCodevalidationFailed(error); })

rather than an empty catching. Are you sure this process may throw something? If you're sure that your helper would not throw anything after all, you can remove this catch.